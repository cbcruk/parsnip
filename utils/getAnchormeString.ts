import anchorme from 'anchorme'

export function getAnchormeString(input: string) {
  return anchorme({
    input,
    options: {
      attributes: {
        target: '_blank',
        rel: 'noreferrer noopener',
      },
    },
  })
}
