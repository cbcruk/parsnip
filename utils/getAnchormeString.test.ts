import { expect, test } from 'vitest'
import { getAnchormeString } from './getAnchormeString'

test('anchorme', () => {
  const result = getAnchormeString('https://google.com')

  expect(result).toMatchSnapshot()
})
