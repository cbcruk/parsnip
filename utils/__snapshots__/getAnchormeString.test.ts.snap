// Vitest Snapshot v1

exports[`anchorme 1`] = `"<a target=\\"_blank\\"  rel=\\"noreferrer noopener\\" href=\\"https://google.com\\">https://google.com</a>"`;
